sim/pokemon.ts

```diff
@@ -517,7 +517,7 @@ export class Pokemon {
 		const health = this.getHealth();
 		let details = this.details;
 		if (this.illusion) {
-			details = this.illusion.getUpdatedDetails(this.battle.ruleTable.has('illusionlevelmod') ? this.level : undefined);
```

Comment:
Is it me or is the problem that we're getting the details using the Illusion user's level if the illusion level mod is active but we want the reverse i.e. if the illusion level mod is active then we want to use the victim's level?